### PR TITLE
Emit event for calls to ReadyToRunInfo::GetEntryPoint.

### PR DESCRIPTION
**Pull request by brianrob**
Adds a new R2RGetEntryPoint event for use in diagnosing R2R method lookup behavior.  As an example, I used this event to help diagnose #23640.

### CODE REVIEW

**Comment by jkotas** [comment]
All similar methods are wired through `ETW::MethodLog::Send*`. Should this one be wired through that as well, with the same or similar formatting and set of arguments like the other methods?

**Comment by fadimounir** [comment]
LGTM. Not sure what's the difference between the ETW::MethodLog::Send* and Fire* logging, but if the rest of the method logging operations are using the Send* APIs, to Jan's comment maybe this one should too.

**Comment by brianrob** [comment]
Seems reasonable.  I've just posted an updated commit.

@fadimounir, there's not a big difference between the two - mostly just conventions.

**Comment by brianrob** [comment]
Thank you both for the review.  I appreciate it.

File: src/vm/readytoruninfo.cpp [hunk 1]
```diff
@@ -788,6 +793,8 @@ PCODE ReadyToRunInfo::GetEntryPoint(MethodDesc * pMD, PrepareCodeConfig* pConfig
 #endif
     }
 
+done:
```
Comment (jkotas) [inline src/vm/readytoruninfo.cpp]: I still think it would be better to check here.

Comment (brianrob) [reply src/vm/readytoruninfo.cpp]: I just saw your comment about EX_TRY.  What about doing this:

```
    if (ETW_EVENT_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_Context, R2RGetEntryPoint))
    {
        EX_TRY
        {
                SString tNamespace, tMethodName, tMethodSignature;
                pMethodDesc->GetMethodInfo(tNamespace, tMethodName, tMethodSignature);

                FireEtwR2RGetEntryPoint(
                    (UINT64)pMethodDesc,
                    (PCWSTR)tNamespace.GetUnicode(),
                    (PCWSTR)tMethodName.GetUnicode(),
                    (PCWSTR)tMethodSignature.GetUnicode(),
                    pEntryPoint,
                    GetClrInstanceId());

        } EX_CATCH{ } EX_END_CATCH(SwallowAllExceptions);
    }
```

I can still put the check inside of GetEntryPoint if you want, but this should reduce the cost pretty well I think.

Comment (brianrob) [reply src/vm/readytoruninfo.cpp]: On second thought, you're probably right given how often this gets called.

**Comment by sergiy-k** [comment]
@brianrob, this event that you added is very expensive. It really obscures startup time data and investigation when an application is started under PerfView. For example, PowerShell spends about 15% of its startup time firing this event:

image

Is it possible to make it not to be enabled by default?

/cc: @billwert

**Comment by brianrob** [comment]
Sorry @sergiy-k, this fell through my mail.  Yes, I intend to move this so it doesn't get enabled by default.  @billwert and I were discussing this a few weeks back.

**Comment by jkotas** [comment]
Opened https://github.com/dotnet/coreclr/issues/25492

**Pull request merged**